rust/otap-dataflow/crates/query-engine/src/pipeline/planner.rs

```diff
@@ -106,6 +122,141 @@ impl PipelinePlanner {
 
         Ok(vec![Box::new(filter_stage)])
     }
+
+    fn plan_move(
+        &self,
+        move_expr: &MoveTransformExpression,
+    ) -> Result<Vec<Box<dyn PipelineStage>>> {
+        match (move_expr.get_source(), move_expr.get_destination()) {
+            (MutableValueExpression::Source(source), MutableValueExpression::Source(dest)) => {
+                let source = ColumnAccessor::try_from(source.get_value_accessor())?;
+                let dest = ColumnAccessor::try_from(dest.get_value_accessor())?;
+
+                match (source, dest) {
+                    // currently the only type of move transform supported is renaming attributes
+                    (
+                        ColumnAccessor::Attributes(src_attrs_id, src_key),
+                        ColumnAccessor::Attributes(dest_attrs_id, dest_key),
+                    ) => {
+                        // the attributes being renamed must be in the same map. for example doing
+                        // `attributes["x"] = resource.attributes["y"]` is not yet supported
+                        if src_attrs_id != dest_attrs_id {
+                            Err(Error::NotYetSupportedError {
+                                message: "renaming attributes from different maps".into(),
```

Comment:
nit - as this is end-user facing error message, should we make it more explicit say "attribute key rename only supports renaming within the same attributes map; found ... ", similar for others.

Reply:
thanks @lalitb, made this change